scripts/o3de/ExportScripts/export_source_built_project.py

```diff
@@ -201,6 +207,18 @@ def export_standalone_project(ctx: exp.O3DEScriptExportContext,
     global o3de_context
     global o3de_logger
 
+    # Resolve the export config
+    try:
+        if o3de_context is None:
```

Comment:
It did run, because I was getting `o3de_context` as None during my ad-hoc testing

Reply:
Ok, it is None because in the situation where we pass in `--script-help`, [we bypass the o3de_context ](https://github.com/o3de/o3de/blob/5c46d85984eb3e8f048b4768dbaed926c07f65bc/scripts/o3de/o3de/export_project.py#L326-L327) because its not needed normally for the help script. However, with the settings, they can be project specific so we added logic to attempt to get project-specific settings for an overlay. so `o3de_context` is set, but with `None`

Reply:
it looks to me like the code on line 381 will just cause the script to output help and exit, so maybe we should print out an error message here and exit too?  It seems like o3de_context is required for the script to run correctly.

Reply:
it really should be. either you pass in the o3de_context to the function directly to run it,  or you run the whole script  through the export-project cli, which should setup the context.

if we suspect at any point the export process doesnt contain a context, that should be a hard error imo

Reply:
Ah just read your message, ok I'm fine with it for now but we should probably refactor this code eventually so we can print the help message without needing all the export_config logic so we can handle it early on.

Reply:
This was a work around from a while ago to get the help menu for these export scripts. But we can refactor this scenario for the general case